src/components/App/App.jsx

```diff
@@ -0,0 +1,156 @@
+
+    /* const redirectUri =
+      import.meta.env.MODE === "production"
+        ? "https://lupesanchez24.github.io/playlist-frontend/callback" // Production Redirect URI
+        : "http://localhost:3000/callback"; // Localhost Redirect URI */
+
+    const redirectUri =
+      import.meta.env.MODE === "production"
+        ? "https://lupesanchez24.github.io/playlist-frontend/#/callback"
+        : "http://localhost:3000/#/callback";
+
+    const scope = "user-library-read user-library-modify";
+
+    const authUrl = `https://accounts.spotify.com/authorize?response_type=code&client_id=${client_id}&redirect_uri=${encodeURIComponent(
+      redirectUri
+    )}&scope=${scope}&code_challenge=${codeChallenge}&code_challenge_method=S256`;
+
+    console.log("Redirect URI used:", redirectUri);
+
+    window.location.href = authUrl;
+
+    //setIsLoggedIn(true);
+  };
+
+  const handleLogout = () => {
+    localStorage.removeItem("spotify_access_token");
+    localStorage.removeItem("spotify_refresh_token");
+    localStorage.removeItem("spotify_token_expiration");
+    setAccessToken(null);
+    setIsLoggedIn(false);
+    setUserData(null);
+    window.location.href = "/";
+  };
+
+  // Handle the callback after user logs from spotify account
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const authorizationCode = urlParams.get("code");
+
+    if (authorizationCode) {
+      getToken(authorizationCode)
+        .then((accessToken) => {
+          if (accessToken) {
+            console.log("Access token received:", accessToken);
+            localStorage.setItem("spotify_access_token", accessToken);
+            setAccessToken(accessToken); // Update the state for user data and routing
+            setIsLoggedIn(true);
+            navigate("/callback");
+          }
+        })
+        .catch((error) => {
+          console.error("Failed to get access token:", error);
+        });
+    }
+  }, []);
+
+  //this is for access toeken
+  useEffect(() => {
+    // Check for an existing token in localStorage on initial load
+    const accessToken = localStorage.getItem("spotify_access_token");
+    if (accessToken) {
+      setAccessToken(accessToken);
+      setIsLoggedIn(true);
+    } else {
+      setIsLoggedIn(false);
+    }
+    setLoading(false);
+  }, []);
+
+  return (
+    <div className="page">
+      <AuthContext.Provider
+        value={{ accessToken, isLoggedIn, setIsLoggedIn, setAccessToken }}
+      >
+        <div className="page__content">
+          <Header
+            handleLogout={handleLogout}
+            handleSpotifyLogin={handleSpotifyLogin}
+          />
+
+          <Routes>
+            <Route path="/" element={<Main />} />
+            <Route
+              path="/callback"
+              element={<Profile accessToken={accessToken} />}
+            />
+            <Route
+              path="/bookmark-page"
+              element={<Bookmark accessToken={accessToken} />}
```

Comment:
`accessToken={accessToken}` is not needed in the props because you should take it from `AuthContext`